crawler/src/main/java/de/unidisk/dao/TopicScoreDAO.java

```diff
@@ -19,6 +20,11 @@ public ScoredInput queryInput(Input input, Session currentSession) {
                 .orElse(new TopicScore((Topic) input));
     }
 
+    @Override
+    public TopicScore addScore(Input input, double score, SearchMetaData smd){
+        return  (TopicScore) ScoringDAO.super.addScore(input,score,smd);
```

Comment:
Ich finde diese Lösung irgendwie komisch. Wäre es hier nicht besser ScoringDAO mit einem Generic zu versehen, was in TopicDAO dann ein TopicScore zurück gibt und eben im KeyDAO einen KeyScore. Dann musst du das hier nicht so merkwürdig überschreiben, dann doch nutzen und am Ende casten.